plugins/reindex/src/main/java/org/elasticsearch/plugin/reindex/AbstractBulkByScrollRequest.java

```diff
@@ -0,0 +1,222 @@
+        extends ActionRequest<Self> {
+    private static final TimeValue DEFAULT_SCROLL_TIMEOUT = TimeValue.timeValueMinutes(5);
+    private static final int DEFAULT_SIZE = 100;
+
+    /**
+     * The search to be executed.
+     */
+    private SearchRequest source;
+
+    /**
+     * Maximum number of processed documents. Defaults to -1 meaning process all
+     * documents.
+     */
+    private int size = -1;
+
+    /**
+     * Should version conflicts cause aborts? Defaults to false.
+     */
+    private boolean abortOnVersionConflict = false;
+
+    /**
+     * Call refresh on the indexes we've written to after the request ends?
+     */
+    private boolean refresh = false;
+
+    /**
+     * Timeout to wait for the shards on to be available for each bulk request?
+     */
+    private TimeValue timeout = ReplicationRequest.DEFAULT_TIMEOUT;
+
+    /**
+     * Consistency level for write requests.
+     */
+    private WriteConsistencyLevel consistency = WriteConsistencyLevel.DEFAULT;
+
+    public AbstractBulkByScrollRequest() {
+    }
+
+    public AbstractBulkByScrollRequest(SearchRequest source) {
+        this.source = source;
+
+        // Set the defaults which differ from SearchRequest's defaults.
+        source.scroll(DEFAULT_SCROLL_TIMEOUT);
+        source.source(new SearchSourceBuilder());
+        source.source().version(true);
+        source.source().sort(fieldSort("_doc"));
+        source.source().size(DEFAULT_SIZE);
+    }
+
+    /**
+     * `this` cast to Self. Used for building fluent methods without cast
+     * warnings.
+     */
+    protected abstract Self self();
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException e = source.validate();
+        if (source.source().from() != -1) {
+            e = addValidationError("from is not supported in this context", e);
+        }
+        return e;
+    }
+
+    /**
+     * Maximum number of processed documents. Defaults to -1 meaning process all
+     * documents.
+     */
+    public int size() {
+        return size;
+    }
+
+    /**
+     * Maximum number of processed documents. Defaults to -1 meaning process all
+     * documents.
+     */
+    public Self size(int size) {
+        this.size = size;
+        return self();
+    }
+
+    /**
+     * Should version conflicts cause aborts? Defaults to false.
+     */
+    public boolean abortOnVersionConflict() {
+        return abortOnVersionConflict;
+    }
+
+    /**
+     * Should version conflicts cause aborts? Defaults to false.
+     */
+    public Self abortOnVersionConflict(boolean abortOnVersionConflict) {
+        this.abortOnVersionConflict = abortOnVersionConflict;
+        return self();
+    }
+
+    /**
+     * Sets abortOnVersionConflict based on REST-friendly names.
+     */
+    public void conflicts(String conflicts) {
+        switch (conflicts) {
+        case "proceed":
```

Comment:
Is this formatting intended (`case` on same indentation level as enclosing `switch`)?